langchain4j-agentic/src/main/java/dev/langchain4j/agentic/AgenticServices.java

```diff
@@ -586,9 +586,59 @@ private static AgentExecutor createComposedAgentExecutor(Class<?> agentServiceCl
             return new AgentExecutor(AgentInvoker.fromMethod(agent, method), agent);
         }
 
+        Optional<Method> humanInTheLoopMethod = getAnnotatedMethodOnClass(agentServiceClass, dev.langchain4j.agentic.declarative.HumanInTheLoop.class);
+        if (humanInTheLoopMethod.isPresent()) {
+            return createHumanInTheLoopAgent(agentServiceClass, humanInTheLoopMethod, humanInTheLoopMethod.get());
+        }
+
+        if (!agentServiceClass.isInterface()) {
+            Method agenticMethod = nonAiAgentMethod(agentServiceClass);
+            if (agenticMethod != null) {
+                if (agenticMethod.getParameterCount() == 0) {
+                    return agentToExecutor(new AgentAction(() -> invokeStatic(agenticMethod)));
+                }
+                if (agenticMethod.getParameterCount() == 1 && AgenticScope.class.isAssignableFrom(agenticMethod.getParameterTypes()[0])) {
+                    return agentToExecutor(new AgenticScopeAction((agenticScope -> invokeStatic(agenticMethod, agenticScope))));
+                }
+            }
+        }
+
         return null;
     }
 
+    private static AgentExecutor createHumanInTheLoopAgent(Class<?> agentServiceClass, Optional<Method> humanInTheLoopMethod, Method method) {
+        var humanInTheLoop = humanInTheLoopMethod.get().getAnnotation(dev.langchain4j.agentic.declarative.HumanInTheLoop.class);
+        if (method.getParameterCount() != 1) {
+            throw new IllegalArgumentException("Method " + method.getName() + " annotated with @HumanInTheLoopRequest must have exactly one parameter");
```

Comment:
```suggestion
            throw new IllegalArgumentException("Method " + method.getName() + " annotated with @HumanInTheLoop must have exactly one parameter");
```
Perhaps it is worth using `HumanInTheLoop.class.getName()` to make sure potential future renaming will not make this error message invalid?

Reply:
Fixed.